p2p: nil pointer in the pex when seed_mode=true!

```
I[06-28|07:34:29.067] Starting MConnection                         module=p2p peer=206.189.9.180:46656 impl=MConn{206.189.9.180:46656}
I[06-28|07:34:29.067] Added peer                                   module=p2p peer="Peer{MConn{206.189.9.180:46656} a92c97f395eea206d2f20dc280c7bbad759339b8 out}"
I[06-28|07:34:29.067] Dialing peer                                 module=p2p address=4720573152c69457cfa9d62d3d9550b4c4babb71@51.38.190.255:46656
I[06-28|07:34:29.109] Stopping queryMaj23Routine for peer          module=consensus peer="Peer{MConn{88.10.255.250:56742} ca556bbd3bd56efab99800d68f79954e65890633 in}"
I[06-28|07:34:29.209] Stopping peer gracefully                     module=p2p 
I[06-28|07:34:29.209] Stopping Peer                                module=p2p peer=88.10.255.250:56748 impl="Peer{MConn{88.10.255.250:56748} ca556bbd3bd56efab99800d68f79954e65890633 in}"
I[06-28|07:34:29.209] Stopping MConnection                         module=p2p peer=88.10.255.250:56748 impl=MConn{88.10.255.250:56748}
I[06-28|07:34:29.219] Timed out                                    module=consensus dur=884.294544ms height=562426 round=0 step=RoundStepNewHeight
I[06-28|07:34:29.219] enterNewRound(562426/0). Current: 562426/0/RoundStepNewHeight module=consensus height=562426 round=0
I[06-28|07:34:29.219] enterPropose(562426/0). Current: 562426/0/RoundStepNewRound module=consensus height=562426 round=0
I[06-28|07:34:29.287] Stopping gossipDataRoutine for peer          module=consensus peer="Peer{MConn{88.10.255.250:56748} ca556bbd3bd56efab99800d68f79954e65890633 in}"
I[06-28|07:34:29.287] Stopping gossipVotesRoutine for peer         module=consensus peer="Peer{MConn{88.10.255.250:56748} ca556bbd3bd56efab99800d68f79954e65890633 in}"
I[06-28|07:34:29.326] Successful handshake with peer               module=p2p peer=51.38.190.255:46656 peerNodeInfo="NodeInfo{id: 4720573152c69457cfa9d62d3d9550b4c4babb71, moniker: melea-trust Sentry-A, network: gaia-6002 [listen 51.38.190.255:46656], version: 0.20.0 ([amino_version=0.9.9 p2p_version=0.5.0 consensus_version=v1/0.2.2 rpc_version=0.7.0/3 tx_index=on rpc_addr=tcp://0.0.0.0:46657])}"
I[06-28|07:34:29.326] Starting Peer                                module=p2p peer=51.38.190.255:46656 impl="Peer{MConn{51.38.190.255:46656} 4720573152c69457cfa9d62d3d9550b4c4babb71 out}"
I[06-28|07:34:29.326] Starting MConnection                         module=p2p peer=51.38.190.255:46656 impl=MConn{51.38.190.255:46656}
E[06-28|07:34:29.326] Connection failed @ recvRoutine (reading byte) module=p2p peer=51.38.190.255:46656 conn=MConn{51.38.190.255:46656} err=EOF
I[06-28|07:34:29.326] Stopping MConnection                         module=p2p peer=51.38.190.255:46656 impl=MConn{51.38.190.255:46656}
E[06-28|07:34:29.326] Stopping peer for error                      module=p2p peer="Peer{MConn{51.38.190.255:46656} 4720573152c69457cfa9d62d3d9550b4c4babb71 out}" err=EOF
I[06-28|07:34:29.326] Stopping Peer                                module=p2p peer=51.38.190.255:46656 impl="Peer{MConn{51.38.190.255:46656} 4720573152c69457cfa9d62d3d9550b4c4babb71 out}"
I[06-28|07:34:29.326] Added peer                                   module=p2p peer="Peer{MConn{51.38.190.255:46656} 4720573152c69457cfa9d62d3d9550b4c4babb71 out}"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xaf808f]

goroutine 89 [running]:
github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tendermint/p2p/pex.(*PEXReactor).RequestAddrs(0xc42049d900, 0x0, 0x0)
        /root/goApps/src/github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tendermint/p2p/pex/pex_reactor.go:285 +0xdf
github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tendermint/p2p/pex.(*PEXReactor).crawlPeers(0xc42049d900)
        /root/goApps/src/github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tendermint/p2p/pex/pex_reactor.go:631 +0x222
github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tendermint/p2p/pex.(*PEXReactor).crawlPeersRoutine(0xc42049d900)
        /root/goApps/src/github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tendermint/p2p/pex/pex_reactor.go:554 +0x81
created by github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tendermint/p2p/pex.(*PEXReactor).OnStart
        /root/goApps/src/github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tendermint/p2p/pex/pex_reactor.go:137 +0xe5
```

My node crashed on the testnet with this ^^

--- a/p2p/peer_set.go
+++ b/p2p/peer_set.go
@@ -55,8 +55,8 @@ func (ps *PeerSet) Add(peer Peer) error {
 	return nil
 }
 
-// Has returns true iff the PeerSet contains
-// the peer referred to by this peerKey.
+// Has returns true if the set contains the peer referred to by this
+// peerKey, otherwise false.
 func (ps *PeerSet) Has(peerKey ID) bool {
 	ps.mtx.Lock()
 	_, ok := ps.lookup[peerKey]
@@ -64,8 +64,8 @@ func (ps *PeerSet) Has(peerKey ID) bool {
 	return ok
 }
 
-// HasIP returns true if the PeerSet contains the peer referred to by this IP
-// address.
+// HasIP returns true if the set contains the peer referred to by this IP
+// address, otherwise false.
 func (ps *PeerSet) HasIP(peerIP net.IP) bool {
 	ps.mtx.Lock()
 	defer ps.mtx.Unlock()
@@ -85,7 +85,8 @@ func (ps *PeerSet) hasIP(peerIP net.IP) bool {
 	return false
 }
 
-// Get looks up a peer by the provided peerKey.
+// Get looks up a peer by the provided peerKey. Returns nil if peer is not
+// found.
 func (ps *PeerSet) Get(peerKey ID) Peer {
 	ps.mtx.Lock()
 	defer ps.mtx.Unlock()

--- a/p2p/pex/pex_reactor.go
+++ b/p2p/pex/pex_reactor.go
@@ -77,10 +77,10 @@ type PEXReactor struct {
 	attemptsToDial sync.Map // address (string) -> {number of attempts (int), last time dialed (time.Time)}
 }
 
-func (pexR *PEXReactor) minReceiveRequestInterval() time.Duration {
+func (r *PEXReactor) minReceiveRequestInterval() time.Duration {
 	// NOTE: must be less than ensurePeersPeriod, otherwise we'll request
 	// peers too quickly from others and they'll think we're bad!
-	return pexR.ensurePeersPeriod / 3
+	return r.ensurePeersPeriod / 3
 }
 
 // PEXReactorConfig holds reactor specific configuration data.
@@ -628,7 +628,9 @@ func (r *PEXReactor) crawlPeers() {
 		}
 		// Ask for more addresses
 		peer := r.Switch.Peers().Get(pi.Addr.ID)
-		r.RequestAddrs(peer)
+		if peer != nil {
+			r.RequestAddrs(peer)
+		}
 	}
 }
 

